### ISSUE TITLE
Possible flaw in `TokenFilterContext#skipParentChecks()`

### ISSUE THREAD

**Issue by robotmrv** [FasterXML/jackson-core#898]
There is a code in `com.fasterxml.jackson.core.filter.TokenFilterContext#skipParentChecks`
```java
public void skipParentChecks() {
    _filter = null;
    for (TokenFilterContext ctxt = _parent; ctxt != null; ctxt = ctxt._parent) {
        _parent._filter = null;
    }
}
```
since `_parent` is a field looks like there is typo in the loop body  
`_parent._filter = null;` -> `ctxt._filter = null;`  

This is actual for at least 2.13, 2.14, 2.15 branches

**Comment by cowtowncoder** [comment]
Yes @robotmrv this does look like a flaw. Thank you for reporting this!

Let me see what happens if I try fix -- I am bit hesitant to make change in a patch so probably needs to in 2.15(.0).
This although method appears to be used for optimization and should not change behavior.

**Comment by cowtowncoder** [comment]
Ok no test failures; seems safe enough so merged fix in, to be included in 2.15.0.
(too bad there's no unit test for this but... if there was, but wouldn't have been)
